src/Tests/FileEntityCreationTest.php

```diff
@@ -136,4 +137,72 @@ public function testImageAltTitleFields() {
       $this->assertEqual($value, $created_file->get($field)->value);
     }
   }
+
+  /**
+   * Test archive upload.
+   */
+  public function testArchiveUpload() {
+    $file_storage = \Drupal::service('entity.manager')->getStorage('file');
+    // Create files for the archive.
+    file_unmanaged_save_data($this->randomString(), 'temporary://test_text.txt');
+    file_unmanaged_save_data($this->randomString(), 'temporary://test_png.png');
+    file_unmanaged_save_data($this->randomString(), 'temporary://test_jpg.jpg');
+
+    $text_file_path = file_directory_temp() . '/test_text.txt';
+    $png_file_path = file_directory_temp() . '/test_png.png';
+    $jpg_file_path = file_directory_temp() . '/test_jpg.jpg';
+
+    $archive_path = file_directory_temp() . '/archive.tar.gz';
+    $archiver = new Tar($archive_path);
+    $archiver->add($text_file_path);
+    $archiver->add($png_file_path);
+    $archiver->add($jpg_file_path);
+
+    $edit = [
+      'files[upload]' => $archive_path,
+      'pattern' => '.*jpg|.*gif',
+      'remove_archive' => TRUE,
+    ];
+    $this->drupalGet('admin/content/files/archive');
+    $this->drupalPostForm(NULL, $edit, t('Submit'));
+
+    $this->assertText('Extracted archive.tar.gz and added 1 new files.');
+
+    // Files that match the pattern can be found in the database.
+    $this->assertTrue($file = !empty($file_storage->loadByProperties(['filename' => 'test_jpg.jpg'])));
```

Comment:
Add meaningful messages to asserts. This applies to all asserts in the test. You can mostly use comments that you have above function calls.